### PR TITLE
Docs: Add first Interactions docs

### PR DESCRIPTION
**Pull request by ghengeveld**
Issue: -

## What I did

First draft. Basically a placeholder to make the "Learn how to add interactions to your story" link in the addon work.

### CODE REVIEW

File: docs/essentials/interactions.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+Stories isolate and capture component states in a structured manner. While developing a component, you can quickly cycle through the stories to verify the look and feel. Each story specifies all the inputs required to reproduce a specific state. You can even mock context and API calls. That allows you to handle most use cases of a component. But what about states that require user interaction?
+
+For example, clicking a button to open/close a dialog box, dragging a list item to reorder it or filling out a form to check for validation errors. To test those behaviours, you have to interact with the components as a user would. Interactive stories enable you to automate these interactions using a play function. These are small snippets of code that script out the exact steps a human would take to interact with the component. It's then executed as soon as the story is rendered.
+
+## Powered by Testing Library and Jest
+
+The interactions are written using a Storybook-instrumented versions of Testing Library and Jes. That gives you a familiar developer-friendly syntax to interact with the DOM and make assertions, but with extra telemetry to help with debugging.
```
Comment (yannbf) [inline docs/essentials/interactions.md:15]:
```suggestion
The interactions are written using a Storybook-instrumented versions of Testing Library and Jest. That gives you a familiar developer-friendly syntax to interact with the DOM and make assertions, but with extra telemetry to help with debugging.
```

File: docs/snippets/common/storybook-interactions-play-function.js.mdx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  component: MyForm,
+  argTypes: {
+    onSubmit: { action: true },
+  },
+};
+
+const Template = (args) => <MyForm {...args} />;
+
+const Submitted = Template.bind({});
+Submitted.play = async ({ canvasElement }) => {
```
Comment (yannbf) [inline docs/snippets/common/storybook-interactions-play-function.js.mdx:18]: do we want to have this notation in the main docs example? or use `context.canvasElement` instead? if people want to reuse play functions, they must pass the entire context, which they lose when destructuring.

File: docs/snippets/common/storybook-interactions-play-function.js.mdx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  const canvas = within(canvasElement);
+
+  await userEvent.type(canvas.getByTestId('email'), 'hi@example.com');
+  await userEvent.type(canvas.getByTestId('password'), 'supersecret');
+  await userEvent.click(canvas.getByRole('button'));
+
+  await waitFor(() => expect(args.onSubmit).toHaveBeenCalled());
```
Comment (yannbf) [inline docs/snippets/common/storybook-interactions-play-function.js.mdx:25]: this args is being used but not defined in the play function signature

Comment (yannbf) [reply docs/snippets/common/storybook-interactions-play-function.js.mdx:25]: also, shouldn't `expect` always be awaited?

File: docs/snippets/common/storybook-interactions-play-function.js.mdx (hunk 2 above)
Comment (ghengeveld) [reply docs/snippets/common/storybook-interactions-play-function.js.mdx:18]:
```suggestion
Submitted.play = async ({ args, canvasElement }) => {
```

File: docs/snippets/common/storybook-interactions-play-function.js.mdx (hunk 3 above)
Comment (ghengeveld) [reply docs/snippets/common/storybook-interactions-play-function.js.mdx:25]: `waitFor` takes an async function, aka a function that returns a Promise. `expect` returns a Promise, so there's no need to wrap it in an async/await block, we can just return the result of expect directly. Of course this only works if you only want to `waitFor` a single thing.

**Pull request merged**